specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1170 @@
+              hint-toolTip = "Decline all"/>
+      </actions>
+</toast>
+```
+
+The ToolTip is useful for buttons with icons and no content. Users can hover over the button with
+the cursor and a text will display describing what the button does.
+
+![Button ToolTip Example](ButtonToolTipExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Content")
+    .AddTextBox(L"textBox", L"Reply")
+    .AddButton(AppNotificationButton()
+        .AddArgument(L"action", L"textReply")
+        .SetInputId(L"textBox")
+        .SetToolTip(L"Send")
+        .SetIcon(winrt::Windows::Foundation::Uri(LR"(Assets/Icons/message.png)")))
+    GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Content</text>
+    </binding>
+  </visual>
+
+    <actions>
+            <input id="textBox" type="text" placeHolderContent="Reply"/>
+            <action
+            content=""
+            imageUri="Assets/Icons/message.png"
+            hint-inputId="textBox"
+            hint-toolTip="Send"
+            arguments="action=textReply"/>
+    </actions>
+</toast>
+```
+
+The placement attribute defines action for a user to interact with when the AppNotification is
+right-clicked. This would typically be used by developers to ask the user for setting changes.
+
+![Button ContextMenu Example](ButtonContextMenuExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Content")
+    .AddTextBox(L"textBox", L"Reply")
+    .AddButton(Button(L"Modify app settings")
+        .AddArgument(L"action", L"textReply")
+        .UseContextMenuPlacement())
+    .GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Content</text>
+    </binding>
+  </visual>
+
+    <actions>
+            <input id="textBox" type="text" placeHolderContent="Reply"/>
+            <action
+            placement="contextMenu"
+            content="Modify app settings"
+            imageUri="Assets/Icons/message.png"
+            arguments="action=textReply"/>
+    </actions>
+</toast>
+```
+
+## Limitations
+
+-   hint-toolTip and hint-buttonStyle are only supported for builds 19041 (20H1) and above. Using
+    these attributes on lower builds will not prevent the AppNotification from being constructed,
+    but developers will receive a debug warning that the attributes are not supported.
+
+# Image
+
+The Image component API sets up the xml for the \<image\> element. Images are visual elements that
+are used to enhance an AppNotification.
+
+The image source can be pulled using one of these protocol handlers:
+
+-   http:// or https:// (A web-based image) (unavailable for unpackageds on build 22000)
+-   ms-appx:/// (An image included in the app package) (only for packaged applications)
+-   ms-appdata:///local/ (An image saved to local storage) (only for packaged applications)
+-   file:/// (A local image)
```

Comment:
I believe local file:\ images work only for Unpackaged apps. Please confirm and add the caveat here.